manager/manager.go

```diff
@@ -49,6 +49,9 @@ type Config struct {
 	// ProtoAddr fields will be used to create listeners otherwise.
 	ProtoListener map[string]net.Listener
 
+	// AdvertiseAddr is a map of addresses to advertise, by protocol.
+	AdvertiseAddr map[string]string
```

Comment:
It only makes sense for TCP, but I made this a map to be consistent with `ProtoAddr` and `ProtoListener` above. I can make this a TCP-specific `string` field if that's cleaner.